x-pack/plugins/fleet/server/services/output.ts

```diff
@@ -86,16 +108,20 @@ async function validateLogstashOutputNotUsedInAPMPolicy(
 }
 
 class OutputService {
+  private get encryptedSoClient() {
+    return appContextService.getInternalUserSOClient(fakeRequest);
```

Comment:
nit: Any advantage to constructing a new client each time this is used? IME, this can have some non-trivial overhead that we might want to avoid.

Reply:
I think this has a small overhead as we already create at least one client for each request, it probably be better if we construct one output service with a client on each request, but currently this service is a used as a singleton and it will lead to a non trivial refacto.